src/lib/utils.ts

```diff
@@ -199,4 +201,79 @@ export function getRenderViewportsForOptions(options: any): Array<Record<string,
         ...mobileRenderViewports[constants.MOBILE_OS_ANDROID]
     ];
+}
+
+// Global SIGINT handler
+process.on('SIGINT', () => {
+    if (isPollingActive) {
+        console.log('Fetching results interrupted. Exiting...');
+        isPollingActive = false;
+    } else {
+        console.log('\nExiting gracefully...');
+    }
+    process.exit(0);
+});
+
+// Background polling function
+export async function startPolling(ctx: Context, task: any): Promise<void> {
+    console.log('Fetching results in progress....');
+    isPollingActive = true;
+
+    const intervalId = setInterval(async () => {
+        if (!isPollingActive) {
+            clearInterval(intervalId);
+            return;
+        }
+        
+        try {
+            const resp = await ctx.client.getScreenshotData(ctx.build.id, ctx.build.baseline, ctx.log);
+
+            fs.writeFileSync(ctx.options.fetchResultsFileName, JSON.stringify(resp, null, 2));
+            ctx.log.debug(`Updated results in ${ctx.options.fetchResultsFileName}`);
+
+            if (resp.build.build_status_ind === 'completed' || resp.build.build_status_ind === 'error') {
+                clearInterval(intervalId);
+                console.log(`Fetching results completed. Final results written to ${ctx.options.fetchResultsFileName}`);
+                isPollingActive = false;
+
+
+                // Evaluating Summary
+                let totalScreenshotsWithMismatches = 0;
+                let totalVariantsWithMismatches = 0;
+                const totalScreenshots = Object.keys(resp.screenshots || {}).length;
+                let totalVariants = 0;
+
+                for (const [screenshot, variants] of Object.entries(resp.screenshots || {})) {
+                    let screenshotHasMismatch = false;
+                    let variantMismatchCount = 0;
+
+                    totalVariants += variants.length; // Add to total variants count
+
+                    for (const variant of variants) {
+                        if (variant.mismatch_percentage > 0) {
+                            screenshotHasMismatch = true;
+                            variantMismatchCount++;
+                        }
+                    }
+
+                    if (screenshotHasMismatch) {
+                        totalScreenshotsWithMismatches++;
+                        totalVariantsWithMismatches += variantMismatchCount;
+                    }
+                }
+
+                // Display summary
+                console.log(`\nSummary of Mismatches:`);
+                console.log(`Total Variants with Mismatches: ${totalVariantsWithMismatches} out of ${totalVariants}`);
+                console.log(`Total Screenshots with Mismatches: ${totalScreenshotsWithMismatches} out of ${totalScreenshots}`);
+                console.log(`Branch Name: ${resp.build.branch_name}`);
+                console.log(`Project Name: ${resp.project.name}`);
+                console.log(`Build ID: ${resp.build.build_id}`);
```

Comment:
All above statements can be compressed into single log

Reply:
Use task output here.